src/DehazingCE.cpp

```diff
@@ -485,8 +488,9 @@ void dehazing::AirlightEstimation(const T* src, int width, int height, int strid
             {
                 const auto pos = (j * width + i) * 3;
                 // peak-r, peak-g, peak-b
-                int nDistance = int(sqrt((peak - src[pos]) * (peak - src[pos]) + (peak - src[pos + 1]) * (peak - src[pos + 1])
-                                         + (peak - src[pos + 2]) * (peak - src[pos + 2])));
+                int nDistance = (int)std::sqrt((float)(peak - src[pos]) * (peak - src[pos]) +
+                                               (float)(peak - src[pos + 1]) * (peak - src[pos + 1]) +
+                                               (float)(peak - src[pos + 2]) * (peak - src[pos + 2]));
```

Comment:
Convert to float or not looks no difference, even for 16bit. There seems to be no overflow problem. Maybe compiler has optimized.
But keep this modification for safety.